### DISCUSSION TITLE
Trying to understand gocardless imports - mappings & ibans

### DISCUSSION THREAD

**Discussion by james-prince** [firefly-iii/firefly-iii#8938] [Firefly III]
Hi, started using firefly-iii yesterday and wow, this is just what i was looking for, very impressive software!

I've read the docs, and checked out the discussions & issues and i can't get to an answer (or i'm missing something).

When i use the importer (with gocardless) for 2 accounts, i end up with 2 sets of transactions (1x withdrawal, 1x deposit).

What i would like to do is have these come up as an individual transfer between the accounts.
I see i could possibly change the type to transfer using rules, but then i think i would end up with duplicates, unless i deleted one and converted the other, is that an appropriate solution?

I have seen that there is a mapping functionality, but it seems to just give the option to link variations of my name to a specific account, i do have a lot of accounts and looking back through my data i can see there are cases where 2 different source accounts share the same "Field value".
eg. i have cases where i have transferred from Bank A and Bank B and they both show up as "B SMITH" as the source.

I have filled in the IBAN numbers on all of my accounts, is there some way i can use this data to map the transactions from gocardless?

Edit: just looked at the data from the gocardless API and it appears that no source IBAN number is present, guessing rules are my only option here?

**Comment by JC5** [comment]
Rules are your only option I'm afraid.

**Reply by james-prince** [reply]
No problem, good to know the rules seem like a workable solution.
Is it just pick one transaction to delete and one to convert?

**Reply by JC5** [reply]
Exactly :).

**Comment by dawid-czarnecki** [comment]
This is actually a bank issue IMO. Some banks don't provide IBAN or other account id information for transaction, which is completely stupid and seems like it's not compliant to the PSD2 (I'm not a lawyer so I could be wrong). I'm in contact with Gocardless who asked a bank to provide this field for all transactions (whenever possible) but they claim it's optional based on Berlin group regulation regarding PSD2. I found the document "Commission delegated Regulation (EU) 2018/389":
>  Account servicing payment service providers shall comply with each of the following requirements: (a) they shall provide account information service providers with the same information from designated payment accounts and associated payment transactions made available to the payment service user when directly requesting access to the account information, provided that this information does not include sensitive payment data

I understand it in a way that if the bank shows an IBAN/account id in your bank app, than it should return it via the API.

Gocardless is helpful but maybe if more of us ask for this feature they would be more motivated to push banks to get it.
